File: Medium/1267-Count-Servers-that-Communicate.go
// You are given a map of a server center, represented as a m * n integer matrix grid, 
// where 1 means that on that cell there is a server and 0 means that it is no server. 
// Two servers are said to communicate if they are on the same row or on the same column.

// Return the number of servers that communicate with any other server.

func countServers(grid [][]int) int {
    res := 0
    m, n := len(grid), len(grid[0])
    rows := make([]int, m)
    columns := make([]int, n)
    
    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {
            if grid[i][j] == 1 {
                rows[i]++
                columns[j]++
            }
        }
    }
    
    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {
            if grid[i][j] == 1 && (rows[i] + columns[j] > 2) {
                res++
            }
        }
    }
    return res
}

File: Medium/1267-Count-Servers-that-Communicate.py
# You are given a map of a server center, represented as a m * n integer matrix grid, 
# where 1 means that on that cell there is a server and 0 means that it is no server. 
# Two servers are said to communicate if they are on the same row or on the same column.

# Return the number of servers that communicate with any other server.


def countServers(grid: list[list[int]]) -> int:

    n = len(grid)
    m = len(grid[0])
    rows, columns = [0]*n, [0]*m

    for i in range(n):
        for j in range(m):
            if grid[i][j]:
                rows[i] += 1
                columns[j] += 1
    res = 0
    for i in range(n):
        for j in range(m):
            if grid[i][j] and (rows[i] > 1 or columns[j] > 1):
                res += 1
    return res



grid = [[1,0],[1,1]]
ans = countServers(grid)
print(ans)